pydoctor/epydoc/markup/_types.py

```diff
@@ -0,0 +1,188 @@
+"""
+Render types from L{docutils.nodes.document} objects. 
+
+This module provides yet another L{ParsedDocstring} subclass.
+"""
+
+from typing import Callable, Dict, List, Tuple, Union
+
+from pydoctor.epydoc.markup import DocstringLinker, ParseError, ParsedDocstring, get_parser_by_name
+from pydoctor.node2stan import node2stan
+from pydoctor.napoleon.docstring import TokenType, TypeDocstring
+
+from docutils import nodes
+from twisted.web.template import Tag, tags
+
+class ParsedTypeDocstring(TypeDocstring, ParsedDocstring):
+    """
+    Add L{ParsedDocstring} interface on top of L{TypeDocstring} and 
+    allow to parse types from L{nodes.Node} objects, providing the C{--process-types} option.
+    """
+    _tokens: List[Tuple[Union[str, nodes.Node], TokenType]]
+
+    def __init__(self, annotation: Union[nodes.document, str],
+                 warns_on_unknown_tokens: bool = False, lineno: int = 0) -> None:
+        ParsedDocstring.__init__(self, ())
+        if isinstance(annotation, nodes.document):
+            TypeDocstring.__init__(self, '', warns_on_unknown_tokens)
+
+            _tokens = self._tokenize_node_type_spec(annotation)
+            self._tokens = self._build_tokens(_tokens)
+        else:
+            TypeDocstring.__init__(self, annotation, warns_on_unknown_tokens)
+        
+        
+        # TODO: do we really need the line number here ?
+        self._lineno = lineno
+
+    @property
+    def has_body(self) -> bool:
+        return len(self._tokens)>0
+
+    def to_node(self) -> nodes.document:
+        """
+        Not implemented.
+        """
+        raise NotImplementedError()
```

Comment:
Complementary comment: We actually need to implement a special `to_stan` method for the `ParsedTypeDocstring` class because of the way it handles the types inside braces in a way it could not be done from Docutils nodes. 

I'll add a exhaustive comment in the code.